timely_beliefs/beliefs/__init__.py

```diff
@@ -119,23 +118,22 @@ def number_of_sources(self):
     @property
     def number_of_probabilistic_beliefs(self) -> int:
         """Return the number of beliefs in the BeliefsDataFrame that are probabilistic (more than 1 unique value)."""
-        df = self._obj.for_each_belief(fnc=pd.DataFrame.nunique, dropna=True)
-        return len(df[df > 1].max(axis=1).dropna())
+        return self.number_of_beliefs - self.number_of_deterministic_beliefs
 
     @property
     def number_of_deterministic_beliefs(self) -> int:
         """Return the number of beliefs in the BeliefsDataFrame that are deterministic (1 unique value)."""
-        return len(self._obj) - self.number_of_probabilistic_beliefs
+        return self._obj.probabilistic_depth_count.get(1, 0)
 
     @property
     def percentage_of_probabilistic_beliefs(self) -> float:
         """Return the percentage of beliefs in the BeliefsDataFrame that are probabilistic (more than 1 unique value)."""
-        return self.number_of_probabilistic_beliefs / self.number_of_beliefs
+        return 1 - self.percentage_of_deterministic_beliefs
 
     @property
     def percentage_of_deterministic_beliefs(self) -> float:
         """Return the percentage of beliefs in the BeliefsDataFrame that are deterministic (1 unique value)."""
-        return 1 - self.number_of_probabilistic_beliefs / self.number_of_beliefs
+        return self.number_of_deterministic_beliefs / self.number_of_beliefs
```

Comment:
In case of an empty bdf I think calling `percentage_of_deterministic_beliefs` will raise a `ZeroDivisionError` because `self.number_of_beliefs = 0`. With that, what would a valid value for the `percentage_of_deterministic_beliefs` in that event? 0 or 1...?